src/components/bookmark-list-container.ts

```diff
@@ -1,30 +1,33 @@
 import { LitElement, html } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
 import { DatabaseService } from '../services/database';
-import { SyncService } from '../services/sync-service';
-import { FaviconService } from '../services/favicon-service';
-import type { BookmarkListContainerState } from '../types';
+import { SyncController } from '../controllers/sync-controller';
+import { FaviconController } from '../controllers/favicon-controller';
+import type { LocalBookmark } from '../types';
 import './bookmark-list';
 
 @customElement('bookmark-list-container')
 export class BookmarkListContainer extends LitElement {
-  @state() private containerState: BookmarkListContainerState = {
+  @state() private containerState: {
+    bookmarks: LocalBookmark[];
+    isLoading: boolean;
+    bookmarksWithAssets: Set<number>;
+  } = {
     bookmarks: [],
     isLoading: true,
-    isSyncing: false,
-    syncProgress: 0,
-    syncTotal: 0,
-    syncedBookmarkIds: new Set<number>(),
-    faviconCache: new Map<number, string>(),
     bookmarksWithAssets: new Set<number>(),
   };
 
-  private syncService: SyncService | null = null;
-  private faviconObserver: IntersectionObserver | null = null;
+  // Reactive controllers
+  private syncController = new SyncController(this, {
+    onBookmarkSynced: this.handleBookmarkSynced.bind(this),
+    onSyncCompleted: this.handleSyncCompleted.bind(this),
+  });
+
+  private faviconController = new FaviconController(this);
 
   override connectedCallback() {
     super.connectedCallback();
-    this.initializeServices();
     this.loadBookmarks();
     
     // Listen for external sync-requested events for backward compatibility
```

Comment:
What is the backwards compatibility need here? We should be updating all of the impacted code in this PR, so there shouldn't be a need to preserve any backwards compatibility. If this isn't needed, remove it and the code it calls (if that code isn't used by other code paths).